### PR TITLE
#8077 Client keep alive fix

### LINKED ISSUE #8077: 4.0.1: WebClient HTTP keep-alive support appears to be broken
## Environment Details
* Helidon Version: 4.0.0 and 4.0.1
* Helidon SE 
* JDK version: java 21.0.1 2023-10-17 LTS
* OS: MacOS

----------

## Problem Description

The Helidon 4 WebClient HTTP keep-alive support appears to be broken. Even with `keepAlive(true)` the client does not appear to reuse connections.

## Steps to reproduce

See the attached test case: [client.zip](https://github.com/helidon-io/helidon/files/13480592/client.zip)

This is based on the stand-alone webclient example that has been modified to set `keepAlive(true)` and to perform 10 GET requests sequentially.  After the test is run (see README) netstat shows 10 connections in TIME_WAIT instead of one:

```
tcp4       0      0  127.0.0.1.56044        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56045        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56046        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56047        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56048        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56049        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56050        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56051        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56052        127.0.0.1.8080         TIME_WAIT
tcp4       0      0  127.0.0.1.56053        127.0.0.1.8080         TIME_WAIT
```

### PR DESCRIPTION
**Pull request by danielkec**
Fixes #8077 and #8110

### CODE REVIEW

File: webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java [hunk 1]
```diff
@@ -96,7 +97,7 @@ class Http1ClientResponseImpl implements Http1ClientResponse {
         this.whenComplete = whenComplete;
 
         if (responseHeaders.contains(HeaderNames.CONTENT_LENGTH)) {
```
Comment (tomas-langer) [inline webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java:99]: there is a method `contentLength()` on server response headers that can be used

File: webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java [hunk 2]
```diff
@@ -166,10 +167,10 @@ public void close() {
                     } else if (entityLength == ENTITY_LENGTH_CHUNKED) {
                         if (hasTrailers) {
                             readTrailers();
```
Comment (tomas-langer) [inline webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java:169]: As far as I understand this, we should always check for `entityFullyRead` - and if not read, we should either consume the entity (e.g. throw it away), or close the connection (as we did originally).
Only when fully read should we release the connection (regardless of chunked encoding or content lenght). The only difference is that with chunked we may also need to read trailers...

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java (hunk 1 above)
Comment (danielkec) [reply webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java]: thx!

File: webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java (hunk 2 above)
Comment (danielkec) [reply webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java]: Fixed and trailered

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**